pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarRecordCursor.java

```diff
@@ -589,42 +588,55 @@ public boolean advanceNextPosition() {
+                    break;
+                case "__event_time__":
                     currentRowValuesMap.put(columnHandle, PulsarFieldValueProviders.timeValueProvider(
                             this.currentMessage.getEventTime(), this.currentMessage.getEventTime() == 0));
-                } else if (PulsarInternalColumn.PUBLISH_TIME.getName().equals(columnHandle.getName())) {
+                    break;
+                case "__publish_time__":
                     currentRowValuesMap.put(columnHandle, PulsarFieldValueProviders.timeValueProvider(
                             this.currentMessage.getPublishTime(), this.currentMessage.getPublishTime() == 0));
-                } else if (PulsarInternalColumn.MESSAGE_ID.getName().equals(columnHandle.getName())) {
+                    break;
+                case "__message_id__":
                     currentRowValuesMap.put(columnHandle, bytesValueProvider(
                             this.currentMessage.getMessageId().toString().getBytes()));
-                } else if (PulsarInternalColumn.SEQUENCE_ID.getName().equals(columnHandle.getName())) {
+                    break;
+                case "__sequence_id__":
                     currentRowValuesMap.put(columnHandle, longValueProvider(this.currentMessage.getSequenceId()));
-                } else if (PulsarInternalColumn.PRODUCER_NAME.getName().equals(columnHandle.getName())) {
+                    break;
+                case "__producer_name__":
                     currentRowValuesMap.put(columnHandle,
                             bytesValueProvider(this.currentMessage.getProducerName().getBytes()));
-                } else if (PulsarInternalColumn.KEY.getName().equals(columnHandle.getName())) {
+                    break;
+                case "__key__":
                     String key = this.currentMessage.getKey().orElse(null);
                     currentRowValuesMap.put(columnHandle, bytesValueProvider(key == null ? null : key.getBytes()));
-                } else if (PulsarInternalColumn.PROPERTIES.getName().equals(columnHandle.getName())) {
+                    break;
+                case "__properties__":
                     try {
                         currentRowValuesMap.put(columnHandle, bytesValueProvider(
                                 new ObjectMapper().writeValueAsBytes(this.currentMessage.getProperties())));
                     } catch (JsonProcessingException e) {
                         throw new RuntimeException(e);
                     }
-                } else {
-                    throw new IllegalArgumentException("unknown internal field " + columnHandle.getName());
-                }
+                    break;
+                default:
+                    try {
+                        throw new IllegalArgumentException("unknown internal field " + columnHandle.getName());
+                    } finally {
+                        break;
+                    }
```

Comment:
```suggestion
                    throw new IllegalArgumentException("unknown internal field " + columnHandle.getName());
```

Also, I don't think the refactor from if-else to switch is necessary. Please revert or explain why you do it.

Reply:
It was because I saw this 9224 issue, the third Internal Column decode optimize
https://github.com/apache/pulsar/issues/9224

Reply:
Thanks for pointing it out. SGTM.